daemon/main.c

```diff
@@ -354,6 +354,16 @@ main(
         pthread_ret = pthread_attr_setstacksize(&pthread_attr, XHA_THREAD_STACKSIZE);
         if (pthread_ret == 0)
         {
+            pthread_ret = pthread_attr_setinheritsched(&pthread_attr, PTHREAD_INHERIT_SCHED);
+            if (pthread_ret)
+            {
+                log_message(MTC_LOG_WARNING, "pthread_attr_setinheritsched failed. (%d)\n", pthread_ret);
```

Comment:
If we don't expect xha to work correctly without these pthread attributes set, should these warnings be hard errors instead?

Reply:
XHA can't work without these attributes in a big pool. But it works well in a small pool event though these attributes fail to set. So would errors be too strict?

Reply:
Do we expect these calls to fail in our environment? I we don't expect them to fail these could be hard errors such that we detect them during testing.

Reply:
> Do we expect these calls to fail in our environment?

No. I think the mostly likely time when these calls would fail would be if we break something by updating the dom0 kernel or glibc and that's something we'd definitely prefer to spot during testing.

Reply:
I changed the log level to error and added exiting code when this failure occurs.